firestore/tests/unit/test__helpers.py

```diff
@@ -86,6 +86,116 @@ def test___ne__type_differ(self):
         self.assertIs(geo_pt1.__ne__(geo_pt2), NotImplemented)
 
 
+class TestFieldPath(unittest.TestCase):
+
+    @staticmethod
+    def _get_target_class():
+        from google.cloud.firestore_v1beta1._helpers import FieldPath
+        return FieldPath
+
+    def _make_one(self, *args, **kwargs):
+        klass = self._get_target_class()
+        return klass(*args, **kwargs)
+
+    def test_empty_string_inside_string_fails(self):
+        with self.assertRaises(ValueError):
+            field_path = self._make_one('a..b')
```

Comment:
This is a valid path and should be represented as `a..b`

Reply:
@schmidt-sebastian 
But if we split on dots, wouldn't this return `['a', '', 'b']`?

Reply:
Another reason why len(parts)==1 should not be a special case.

Reply:
Please take a look at Node's `fromDotSeparatedString`: https://github.com/googleapis/nodejs-firestore/blob/d85f33e9b8538f34ac3f2d9a07495bd32a65c325/src/path.js#L326. This is a special entry point that we use internally. The public entry point to FieldPath does not split on dots.

`query.where('a.b', ... )` calls `fromDotSeparatedString` and represents `{a : b: ... }` . `query.where(new FieldPath('a.b'), ... )` represents `{ a.b : ...}`